README.md

```diff
@@ -0,0 +1,88 @@
+```sh
+  ################################################
+  ##              ###                            ##
+  ##             ##                              ##
+  #######       ##      ##       ##     ###     ##
+      ##       ##        #########     ##      ##
+     ##       ##               ##           ###
+    ##        #######         ##           ##
+   ##               ##       ##             ##
+  ##                       ###      ###      ##
+  ##################################  #########
+
+     Z S E R I O  C + +  R E F L E C T I O N
+```
+
+# About
+
+Zsr provides a c++ runtime reflection layer for [zserio](https://github.com/ndsev/zserio) generated c++ modules.
+
+# Building
+
+```sh
+mkdir build && cd build
+cmake -DZSERIO_RT_DIR=<path/to/zserio/compiler/extensions/cpp/runtime/src> ..
+cmake --build .
+```
+
+```sh
+ctest .
+```
+
+# Usage
+
+## CMake
+
+In your CMake project, call the `add_zserio_module` function to generate and
+build a zserio c++ module with reflection metadata.
+
+```cmake
+add_zserio_module(<module-name>
+    ROOT <zserio module source root>
+    ENTRY <entrypoint zserio source. e.G. "main.zs">
+    TOP_LEVEL_PKG <optional: top level package name>)
+```
+
+This will create two CMake targets `<module-name>` and `<module-name>-reflection` (the later links the first).
+To link against the generated zserio module, just link against `<module-name>-reflection`
+
+## Reflection
+
+Zsr stores its top-level reflection metadata in a global list that can be accessed by calling `zsr::packages()`.
+Metadata types and their hiararchy can be found in [runtime/zsr/types.hpp](./runtime/zsr/types.hpp).
+
+There are various utility functions for finding metadata by name: `zsr::find<...>(...)` which take a list of packages to
```

Comment:
This reads a bit weird: "Various utility functions: One function explicitely named, then a list of headers". Might be better to list all of the utility functions nested under their respective headers... 🤔

Reply:
I think `find` can also take a bit more explanation. Which meta-types can be found, how can they be used....